homeassistant/components/switchbot_cloud/__init__.py

```diff
@@ -109,6 +112,18 @@ async def async_setup_entry(hass: HomeAssistant, config: ConfigEntry) -> bool:
     await gather(
         *[coordinator.async_refresh() for coordinator in coordinators_by_id.values()]
     )
+
+    list_devices_service = make_list_devices(api)
+    hass.services.async_register(
+        DOMAIN,
+        "list_devices",
+        list_devices_service,
+        supports_response=SupportsResponse.ONLY,
+    )
```

Comment:
Why do we need a service to list all the devices? We already have them in Home Assistant right?

Reply:
> we do not have all the device types that the SwitchBot API provides

You mean implemented in HA?

Reply:
Yeah, not yet implemented in HA